screens/ChatsScreen.js

```diff
@@ -53,13 +59,48 @@ class ChatsScreen extends React.Component {
     };
 
     componentDidMount() {
-        if (!this.props.currentUser) {
+        const {
+            currentUser,
+            setCurrentUser,
+            createSocketConnection,
+            fetchChats
+        } = this.props;
+
+        if (!currentUser) {
             this.getUser().then((user) =>
-                this.props.setCurrentUser(user)
+                setCurrentUser(user)
             );
         }
 
-        this.props.fetchChats();
+        createSocketConnection();
+        fetchChats();
+    }
+
+    componentDidUpdate(prevProps) {
+        const { error, navigation, socket } = this.props;
+
+        if (error && prevProps.error !== error && error.status === 401) {
+            if (socket) {
+                socket.disconnect();
+            }
+            Toast.show({
+                text: `${error.message}; you must login again.`,
+                buttonText: "Okay",
+                type: "danger",
+                duration: 2000,
+                onClose: () => {
+                    this.clearAsyncStorage();
+                    navigation.navigate('Auth')
+                }
+            });
```

Comment:
This is kind of messy and I'm not sure how best to handle it alternatively. The current flow is that only AsyncStorage is checked for on the AuthLoadingScreen to verify that a user is authenticated, so that redirects a user to the Chats screen even if the token isn't necessarily valid (i.e., even if it's already expired). So I'm just checking for a 401 error with the fetchChats call (GET request to `/chats`), displaying a toast if so, and navigating back to Auth (i.e., the Login screen).

Reply:
that works, another way is if we had determined the expiry date and then when the user logs in, set an expiration date of current datetime + expiration in asyncstorage, and then check for the current datetime being less than that in authloadingscreen and clear it if so.

Reply:
Ah yeah, I thought of this idea too. I was concerned about being off by some ms or s though since the time at which the token is generated on the backend is different from the time the token reaches the client, if that makes sense. I think I found an alternative though using [jwt-decode](https://github.com/auth0/jwt-decode) on the client to get and store the exp time (just decoding works since the exp time isn't an encrypted value). 🎉